core/templates/dev/head/domain/exploration/EditableExplorationBackendApiService.js

```diff
@@ -0,0 +1,161 @@
+        UrlInterpolationService.interpolateUrl(
+        EDITABLE_EXPLORATION_DATA_URL_TEMPLATE, {
+          exploration_id: explorationId
+        })
+      );
+
+      var putData = {
+        version: explorationVersion,
+        commit_message: commitMessage,
+        change_list: changeList
+      };
+      $http.put(editableExplorationDataUrl, putData).then(function(response) {
+        // The returned data is an updated exploration dict.
+        var exploration = angular.copy(response.data);
+
+        // Delete from the ReadOnlyExplorationBackendApiService's cache
+        // As the two versions of the data (learner and editor) now differ
+        ReadOnlyExplorationBackendApiService.deleteExplorationFromCache(
+          explorationId, exploration);
+
+        if (successCallback) {
+          successCallback(exploration);
+        }
+      }, function(errorResponse) {
+        if (errorCallback) {
+          errorCallback(errorResponse.data);
+        }
+      });
+    };
+
+    var _deleteExploration = function(
+        explorationId, role, successCallback, errorCallback) {
+      var editableExplorationDataUrl = (
+        UrlInterpolationService.interpolateUrl(
+        EDITABLE_EXPLORATION_DATA_URL_TEMPLATE, {
+          exploration_id: explorationId
+        })
+      );
+
+      params = {};
+      if (role) {
+        params.role = JSON.stringify(role);
+      }
+
+      $http['delete'](editableExplorationDataUrl, {
+        params: params
+      }).then(function() {
+        // Delete item from the ReadOnlyExplorationBackendApiService's cache
+        ReadOnlyExplorationBackendApiService.deleteExplorationFromCache(
+          explorationId);
+        if (successCallback) {
+          successCallback({});
+        }
+      }, function(errorResponse) {
+        if (errorCallback) {
+          errorCallback(errorResponse.data);
+        }
+      });
+    };
+
+    return {
+      fetchExploration: function(explorationId, applyDraft) {
+        return $q(function(resolve, reject) {
+          _fetchExploration(explorationId, applyDraft, resolve, reject);
+        });
+      },
+
+      /**
+       * Updates an exploration in the backend with the provided exploration
+       * ID. The changes only apply to the exploration of the given version
+       * and the request to update the exploration will fail if the provided
+       * exploration version is older than the current version stored in the
+       * backend. Both the changes and the message to associate with those
+       * changes are used to commit a change to the exploration.
+       * The new exploration is passed to the success callback,
+       * if one is provided to the returned promise object. Errors are passed
+       * to the error callback, if one is provided. Finally, if the update is
+       * successful, the returned exploration will be cached within the
+       * ReadOnlyExplorationBackendApiService to ensure the cache is
```

Comment:
Is this true? Isn't the exploration removed from the read only backend api service if it's updated?